pkg/cmd/cluster/connect.go

```diff
@@ -435,20 +436,36 @@ func getUserAndPassword(clusterDef *appsv1alpha1.ClusterDefinition, secrets *cor
 	}
 
 	// now, we only use the first secret
-	var secret corev1.Secret
+	var secret *corev1.Secret
 	for i, s := range secrets.Items {
 		if strings.Contains(s.Name, "conn-credential") {
-			secret = secrets.Items[i]
+			secret = &secrets.Items[i]
 			break
 		}
 	}
-	user, err = getSecretVal(&secret, "username")
+	// 0.8 API connect
+	if secret == nil && compDef != nil {
+		for i := range compDef.Spec.SystemAccounts {
+			if compDef.Spec.SystemAccounts[i].InitAccount {
+				for j, s := range secrets.Items {
+					if strings.Contains(s.Name, compDef.Spec.SystemAccounts[j].Name) {
+						secret = &secrets.Items[j]
+						break
+					}
+				}
+			}
+			if secret != nil {
+				break
+			}
+		}
+	}
+	user, err = getSecretVal(secret, "username")
 	if err != nil {
 		return user, password, err
 	}
 
 	passwordKey := getPasswordKey(clusterDef.Spec.ConnectionCredential)
```

Comment:
it it not guaranteed that conn-credential secret, and sysacc secret use 'passowrd' as key for password